src/core/core.controller.js

```diff
@@ -322,10 +323,11 @@ module.exports = function(Chart) {
 			}
 
 			// Clip out the chart area so that anything outside does not draw. This is necessary for zoom and pan to function
-			this.chart.ctx.save();
-			this.chart.ctx.beginPath();
-			this.chart.ctx.rect(this.chartArea.left, this.chartArea.top, this.chartArea.right - this.chartArea.left, this.chartArea.bottom - this.chartArea.top);
-			this.chart.ctx.clip();
+			var context = this.chart.ctx;
```

Comment:
It's only when you have ctx and this.ctx or ticks and this.ticks that refactoring really fails. They could be pointing to the same thing or an options ticks. Since we are not using typescript which has way better tooling, it fails pretty quick